tools/rule-notices.ts

```diff
@@ -0,0 +1,109 @@
+import plugin from '../src';
+
+enum MESSAGE_TYPE {
+  CONFIGS = 1,
+  DEPRECATED = 2,
+  FIXABLE = 3,
+  HAS_SUGGESTIONS = 4,
+}
+
+export const MESSAGES = {
+  [MESSAGE_TYPE.CONFIGS]:
+    '💼 This rule is enabled in the following [configs](https://github.com/jest-community/eslint-plugin-jest#shareable-configurations):',
+  [MESSAGE_TYPE.DEPRECATED]: '❌ This rule is deprecated.',
+  [MESSAGE_TYPE.FIXABLE]:
+    '🔧 This rule is automatically fixable using the `--fix` [option](https://eslint.org/docs/latest/user-guide/command-line-interface#--fix) on the command line.',
+  [MESSAGE_TYPE.HAS_SUGGESTIONS]:
+    '💡 This rule provides [suggestions](https://eslint.org/docs/developer-guide/working-with-rules#providing-suggestions) that can be applied manually.',
+};
+
+type RuleName = keyof typeof plugin.rules;
+type Rule = typeof plugin.rules[RuleName];
```

Comment:
I'd rather not have the `@ts-ignore` below instead of this as I don't think there's a lot of value vs using `string` & co; it might be possible to make it work but I'd have to play around with the types a bit (which I don't have the time for right now) so if you can't make it work lets just go with `string`.

Reply:
This `RuleName` type I have already evaluates to a `string`. I went ahead and removed this type since it's not really necessary, but this does not solve the `@ts-ignore` issue below.

```
Element implicitly has an 'any' type because expression of type '`jest/${string}`' can't be used to index type 'Record<string, RuleLevel> | { 'jest/no-alias-methods': string; 'jest/prefer-to-be': string; 'jest/prefer-to-contain': string; 'jest/prefer-to-have-length': string; }'.ts(7053)
```

This could be solved if we had a static type containing all the possible rule names, but I don't think we have that, and having to define a type containing all the rule names would be a maintenance burden, so it seems best to just use `@ts-ignore` for this situation.